service/src/main/java/com/epam/esm/service/CertificateService.java

```diff
@@ -0,0 +1,86 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.exception.EntityAlreadyExistsException;
+import com.epam.esm.exception.EntityNotFoundException;
+import com.epam.esm.exception.InvalidEntityException;
+import com.epam.esm.repository.SearchCriteria;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Interface that provides functionality for manipulating {@link Certificate} entity.
+ *
+ * @author Geller Egor
+ */
+@Service
+public interface CertificateService {
```

Comment:
I would suggest to create more common interface EntityService<T> with type parameter and declare CRUD method there.

Reply:
Please update

Reply:
fixed